interactive_engine/assembly/src/bin/graphscope/giectl

```diff
@@ -330,37 +338,89 @@ create_gremlin_instance_on_k8s() {
   log "Expose gremlin server."
   timeout_seconds=60
   # random from range [50001, 51000) for interactive engine
-  frontend_external_port=$(( ((RANDOM<<15)|RANDOM) % 1000 + 50000 ))
+  frontend_external_gremlin_port=$(( ((RANDOM<<15)|RANDOM) % 1000 + 50000 ))
+  frontend_external_cypher_port=$(( ((RANDOM<<15)|RANDOM) % 1000 + 50000 ))
   frontend_deployment_name=$(kubectl get deployment -lapp.kubernetes.io/component=frontend,app.kubernetes.io/instance=${instance_id},app.kubernetes.io/engine_selector=${engine_selector} -o jsonpath='{.items[*].metadata.name}')
   if [ "${GREMLIN_EXPOSE}" = "LoadBalancer" ]; then
-    kubectl expose deployment ${frontend_deployment_name} --name=gremlin-${object_id} --port=${frontend_external_port} \
-      --target-port=${frontend_port} --type=LoadBalancer 1>/dev/null 2>&1
+    # gremlin
+    kubectl expose deployment ${frontend_deployment_name} --name=gremlin-${object_id} --port=${frontend_external_gremlin_port} \
+      --target-port=${frontend_gremlin_port} --type=LoadBalancer 1>/dev/null 2>&1
     [ $? -eq 0 ] || exit 1
     wait_period_seconds=0
     while true
     do
-      external_ip=$(kubectl get service gremlin-${object_id} -ojsonpath='{.status.loadBalancer.ingress[0].ip}') || true
-      if [ -n "${external_ip}" ]; then
+      external_gremlin_ip=$(kubectl get service gremlin-${object_id} -ojsonpath='{.status.loadBalancer.ingress[0].ip}') || true
+      if [ -n "${external_gremlin_ip}" ]; then
         break
       fi
       wait_period_seconds=$(($wait_period_seconds+5))
       if [ ${wait_period_seconds} -gt ${timeout_seconds} ];then
-        echo "Get external ip of ${GREMLIN_EXPOSE} failed."
+        echo "Get gremlin external ip of ${GREMLIN_EXPOSE} failed."
+        break
+      fi
+      sleep 5
+    done
+    # cypher
+    kubectl expose deployment ${frontend_deployment_name} --name=cypher-${object_id} --port=${frontend_external_cypher_port} \
+      --target-port=${frontend_cypher_port} --type=LoadBalancer 1>/dev/null 2>&1
+    [ $? -eq 0 ] || exit 1
+    wait_period_seconds=0
+    while true
+    do
+      external_cypher_ip=$(kubectl get service cypher-${object_id} -ojsonpath='{.status.loadBalancer.ingress[0].ip}') || true
+      if [ -n "${external_cypher_ip}" ]; then
+        break
+      fi
+      wait_period_seconds=$(($wait_period_seconds+5))
+      if [ ${wait_period_seconds} -gt ${timeout_seconds} ];then
+        echo "Get cypher external ip of ${GREMLIN_EXPOSE} failed."
```

Comment:
Consider extract this logic into a function, since they are so similar, to alleviate the maintenance overhead.